node/store.go

```diff
@@ -14,65 +14,75 @@ import (
 	"context"
 
 	"github.com/sourcenetwork/defradb/datastore"
-	"github.com/sourcenetwork/defradb/datastore/badger/v4"
-	"github.com/sourcenetwork/defradb/datastore/memory"
 )
 
+type StoreType string
+
+const (
+	// The Go-enum default StoreType.
+	//
+	// The actual store type that this resolves to depends on the build target.
+	DefaultStore StoreType = ""
+)
+
+// storeConstructors is a map of [StoreType]s to store constructors.
+//
+// Is is populated by the `init` functions in the runtime-specific files - this
+// allows it's population to be managed by build flags.
+var storeConstructors = map[StoreType]func(ctx context.Context, options *StoreOptions) (datastore.Rootstore, error){}
+
 // StoreOptions contains store configuration values.
 type StoreOptions struct {
-	path             string
-	inMemory         bool
-	defraStore       bool
-	valueLogFileSize int64
-	encryptionKey    []byte
+	store               StoreType
+	badgerPath          string
+	badgerFileSize      int64
+	badgerEncryptionKey []byte
+	badgerInMemory      bool
 }
 
 // DefaultStoreOptions returns new options with default values.
 func DefaultStoreOptions() *StoreOptions {
 	return &StoreOptions{
-		inMemory:         false,
-		valueLogFileSize: 1 << 30,
+		badgerInMemory: false,
+		badgerFileSize: 1 << 30,
 	}
 }
 
 // StoreOpt is a function for setting configuration values.
 type StoreOpt func(*StoreOptions)
 
-// WithInMemory sets the in memory flag.
-func WithInMemory(inMemory bool) StoreOpt {
+// WithStoreType sets the store type to use.
+func WithStoreType(store StoreType) StoreOpt {
 	return func(o *StoreOptions) {
-		o.inMemory = inMemory
+		o.store = store
 	}
 }
 
-// WithDefraStore sets the defra store flag.
-//
-// Setting this to true will result in the defra node being created with
-// the a custom defra implementation of the rootstore instead of badger.
-func WithDefraStore(defraStore bool) StoreOpt {
+// WithBadgerInMemory sets the badger in memory option.
+func WithBadgerInMemory(enable bool) StoreOpt {
 	return func(o *StoreOptions) {
-		o.defraStore = defraStore
+		o.badgerInMemory = enable
 	}
 }
 
-// WithPath sets the datastore path.
-func WithPath(path string) StoreOpt {
+// WithBadgerPath sets the badger datastore path.
+func WithBadgerPath(path string) StoreOpt {
 	return func(o *StoreOptions) {
-		o.path = path
+		o.badgerPath = path
```

Comment:
question: Why have you made these (props and public funcs) store specific?

Reply:
I renamed `badgerPath` to `path`.